app/controllers/submissions_controller.rb

```diff
@@ -218,23 +218,61 @@ def collect_and_begin_grading
 
   def collect_all_submissions
     assignment = Assignment.includes(:groupings).find(params[:assignment_id])
-    if assignment.submission_rule.can_collect_now?
+    if assignment.submission_rule.can_collect_all_now?
       submission_collector = SubmissionCollector.instance
       submission_collector.push_groupings_to_queue(assignment.groupings)
-      flash[:success] =
+      success =
           I18n.t('collect_submissions.collection_job_started',
                  assignment_identifier: assignment.short_identifier)
+      render json: { success: success }
     else
-      flash[:error] = I18n.t('collect_submissions.could_not_collect',
+      error = I18n.t('collect_submissions.could_not_collect',
                              assignment_identifier: assignment.short_identifier)
+      render json: { error: error }
     end
-    redirect_to action: 'browse',
-                id: assignment.id
+  end
+
+  def collect_section_submissions
+    assignment = Assignment.includes(:groupings).find(params[:assignment_id])
+    section_ids = params[:sections]
+    errors = Array.new
+    successes = Array.new
+    noSubmissions = Array.new
+    section_ids.each do |id|
+      if !Section.exists?(id)
+        errors.push(I18n.t('collect_submissions.could_not_find_section'))
+        next
+      end
+        
+      if collect_submissions_for_section(id, assignment, errors) > 0
+        successes.push(Section.find(id).name)
+      else
+        noSubmissions.push(Section.find(id).name)
+      end
+    end
+    if successes.length > 0
+      plural = successes.length > 1 ? 's' : ''
```

Comment:
Remove the logic for pluralization and final "and". It's fine to just have section lists that look like

```
Sections collected: s1, s2, s3
```